api/v1/services/AuthService.js

```diff
@@ -65,3 +68,48 @@ module.exports.verify = function(token) {
 			throw new errors.UnprocessableRequestError(message);
 		});
 };
+
+
+/**
+ * Generates a token and deletes all existing tokens
+ * with the same scope.
+ * @param {User} user The user object to create a reset token for.
+ * @param {String} scope The scope to create the token for.
+ * @return {Promise<Bool>} Returns a Promise that resolves to
+ *                         true on a successful token creation.
+ */
+module.exports.generateToken = function(user, scope){
+	var tokenVal = utils.crypto.generateResetToken();
+	var user_id = user.get('id');
+
+	return Token
```

Comment:
This seems a bit confusing. How about we change the following so that it: queries for non-expired tokens (there should only be at most one), deletes any that are not expired (at most one), and then makes a new one no matter what?
Right now we're cleaning out all of the potentially-old tokens on every request, but it's probably best if we make it so that we can `fetchOne` instead of trying to `fetchAll`